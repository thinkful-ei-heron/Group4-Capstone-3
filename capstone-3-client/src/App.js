import React from 'react';
import Register from "./Register/Register";
import {Route, Switch} from "react-router-dom";

function App() {
  return (
    <main className='App'>
      {/* content goes here */}
      <Switch>
          <Route exact to={'/register'}><Register/></Route>
      </Switch>

    </main>
  );
}

export default App;
