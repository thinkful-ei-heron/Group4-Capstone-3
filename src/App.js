import React, {Component} from 'react';
import Register from "./Register/Register";
import {Switch} from "react-router-dom";
import PublicOnlyRoute from "./PublicOnlyRoute/PublicOnlyRoute";
import PrivateOnlyRoute from './PrivateRoute/PrivateRoute';
import LandingPage from "./LandingPage/LandingPage";
import './App.css'
import LoginForm from "./LoginForm/login-form";
import AddForm from "./AddForm/add-form";
import Dashboard from './Dashboard/Dashboard';

export default class App extends Component {
    state = { hasError: false };

    render(){
        const { hasError } = this.state;
        return (
            <div className='App'>
                {hasError && (
                    <p>There was an error! Oh no!</p>
                )}
                <Switch>
                    <PrivateOnlyRoute path={'/home'} component={Dashboard}/>
                    <PrivateOnlyRoute path={'/add'} component={AddForm}/>
                    <PublicOnlyRoute path={'/register'} component={Register}/>
                    <PublicOnlyRoute path={'/login'} component={LoginForm}/>
                    <PublicOnlyRoute exact path={'/'} component={LandingPage}/>
                </Switch>

            </div>
        );
    }
}