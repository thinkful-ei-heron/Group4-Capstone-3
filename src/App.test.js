import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom';
import App from './App'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import {ParallaxProvider} from 'react-scroll-parallax';

Enzyme.configure({ adapter: new Adapter() });

describe(`App component`, () => {
it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <ParallaxProvider>
        <App />
      </ParallaxProvider> 
    </BrowserRouter>,
    div)
  ReactDOM.unmountComponentAtNode(div)
})

it('Displays App when rendered', () => {
  const wrapper = shallow(<App />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
})