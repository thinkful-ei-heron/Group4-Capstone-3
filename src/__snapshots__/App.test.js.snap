// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Displays App when rendered 1`] = `
<div
  className="App"
>
  <PrivateRoute
    component={[Function]}
    exact={true}
    path="/add/tutorial"
  />
  <Switch>
    <PrivateRoute
      component={[Function]}
      path="/breweries"
    />
    <PrivateRoute
      component={[Function]}
      path="/home"
    />
    <PrivateRoute
      component={[Function]}
      path="/add"
    />
    <PublicOnlyRoute
      component={[Function]}
      path="/register"
    />
    <PublicOnlyRoute
      component={[Function]}
      path="/login"
    />
    <PublicOnlyRoute
      component={[Function]}
      exact={true}
      path="/"
    />
  </Switch>
</div>
`;
